base/base/constexpr_helpers.h

```diff
@@ -1,32 +0,0 @@
-#pragma once
-
-#include <type_traits>
-#include <utility>
-
-template <auto Val, decltype(Val)... List>
-inline constexpr bool static_in_v = std::disjunction_v<std::bool_constant<Val == List>...>;
-
-template <typename Func, typename Arg>
-bool func_wrapper(Func && func, Arg && arg)
```

Comment:
Why did you remove `func_wrapper`?

Reply:
Unneeded as all functors in static_for should return bool (simpler semantics).

Reply:
No, I think that's not OK. It leads to writing unnecessary code (for example, changes in `DictionaryStructure.h`).

Reply:
The only isage is in Dictionary structure, others always return bool. And even in dictionary structure my solution (exit iteration after type discovery) is better than previous (iterate over all types even after match was found).

Reply:
`static_for` is an utility function. It doesn't matter how is it used now. The only important thing is how it'll be used in the future after your change.
By removing `func_wrapper` you say that in the future nobody never will want to iterate over all the elements of the container.
I see that as a bad API design, because you can not guarantee that.

Reply:
Handling void return types introduces additional complexity both to function implementation and to function semantics which won't pay off.

Reply:
1. If I want to iterate over all elements I don't want to check whether I need to continue processing or not. C++ ideology say "you don’t pay for what you don’t use".
2. I think it's possible to solve this problem by one simple `if constexpr` in `static_for`. It'll not make anything complex.
3. I think that anybody who uses `static_for` will expect it to iterate over all element if function returns void type. But you can add as a comment that if that person wants to be able to break the execution of the loop if may use another return type.

Reply:
Ok, I guess you're right here. I'll change the solution to reflect void return types (but won't change any of the existing static_for calls), is it fine for you?